### DISCUSSION TITLE
Binary-less version

### DISCUSSION THREAD

**Discussion by ghost** [Heroic-Games-Launcher/HeroicGamesLauncher#490] [Ideas]
**Is your feature request related to a problem? Please describe.**
Right now heroic comes with Legendary installed by default. Which increases the app's size a lot. It will be reduced with #470 (edit: I meant #467) but still, Legendary adds few MBs.

**Describe the solution you'd like**
A version that doesn't pack the legendary binary and instead uses the one in PATH.

**Describe alternatives you've considered**
We could also (theoretically) call Epic's API directly.

**Additional context**
.

**Comment by Nocccer** [comment]
Fix for #467 will also reduce app's size.

**Comment by flavioislima** [comment]
> **Is your feature request related to a problem? Please describe.**
> Right now heroic comes with Legendary installed by default. Which increases the app's size a lot. It will be reduced with #470 but still, Legendary adds few MBs.
> 
> **Describe the solution you'd like**
> A version that doesn't pack the legendary binary and instead uses the one in PATH.
> 
> **Describe alternatives you've considered**
> We could also (theoretically) call Epic's API directly.
> 
> **Additional context**
> .

But for that we would need to have legendary installed first.
I don't think 7Mb is a huge deal since we wont depend on another software being installed, might be confused and we might not have it on all distros repos as well.
When heroic started I was doing like this but then people were always confused because they needed to install legendary in order to heroic to work. So that's why I started to bundle it.

**Comment by Nocccer** [comment]
@dragonDScript one possible solution for packages could be a python pip legendary-gl dependency.
You can set dependencies in the electron-builder with depends. Tried to set minimum python version need for heroic there but failed. If you familiar with electron-builder and package dependencies try it.

**Comment by flavioislima** [comment]
Let's turn this into a discussion.
